packages/amplify-app/src/index.js

```diff
@@ -38,11 +42,7 @@ function run() {
 // Node version check
 async function checkNodeVersion() {
   const currentNodeVersion = process.versions.node;
-  const semver = currentNodeVersion.split('.');
-  const major = semver[0];
-  const minor = semver[1];
-
-  if (major < 8 || (major === 8 && minor < 12)) {
+  if (semver.satisfies(currentNodeVersion, minNodeVersion)) {
     console.error(
       `You are running Node ${currentNodeVersion}.\n` +
         `Amplify CLI requires Node 8.12.0 or higher. \n` +
```

Comment:
String interpolate the minimum version from the version we read above instead make if part of the string.